Adding Fine-tuning to Get-Started section (#113)
kluster-ai/docs: get-started/fine-tuning/api.md

---
title: Fine-tuning with the API
description: Learn how to programmatically create custom models tailored to your specific tasks by fine-tuning foundation models with your own data using the kluster.ai API.
---

# Fine-tuning with the API

The kluster.ai API lets you automate and integrate fine-tuning into your development workflows. You can create, manage, and monitor fine-tuning jobs directly from your code, making it easy to customize models for your specific needs.

This guide provides a practical overview of the fine-tuning process using the API. It covers the required data format, how to upload your dataset, and how to launch and monitor a fine-tuning job. For a step-by-step walkthrough, see the linked tutorial in the tips below.

## Prerequisites

Before getting started with fine-tuning, ensure you have the following:

> - **kluster.ai account**: Sign up on the [kluster.ai platform](https://platform.kluster.ai/signup){target=_blank} if you do not have one.
> - **kluster.ai API key**: After signing in, go to the [API Keys](https://platform.kluster.ai/apikeys){target=_blank} section and create a new key. For detailed instructions, see the [Get an API key](https://docs.kluster.ai/get-started/get-api-key/){target=_blank} guide.

Reviewer: ```suggestion
--8<-- 'text/kluster-api-onboarding.md'
```

- **Prepared dataset**: You need data formatted according to kluster.ai's requirements for fine-tuning (detailed below).

## Supported models

kluster.ai currently supports fine-tuning for two base models:

- [klusterai/Meta-Llama-3.1-8B-Instruct-Turbo](https://docs.kluster.ai/get-started/models/){target=_blank}
- [klusterai/Meta-Llama-3.3-70B-Instruct-Turbo](https://docs.kluster.ai/get-started/models/){target=_blank} 

!!! info
    You can query the [models endpoint](https://docs.kluster.ai/api-reference/reference/#list-supported-models){target=_blank} in the API and filter for the tag "fine-tunable."

## Fine-tuning workflow

> The process of fine-tuning a model using the kluster.ai API involves several key steps:

Reviewer: I would add the steps first, like we do with Batch, so that the readers understand them before doing a deep dive into each. Similar to -> https://docs.kluster.ai/get-started/start-building/batch/#batch-job-workflow-overview

    },
    {
      "role": "user",
      "content": "Create a configuration for a web application with name 'TaskMaster', version 1.2.0, and environment set to development."
    },
    {
      "role": "assistant",
      "content": "{\n  \"application\": {\n    \"name\": \"TaskMaster\",\n    \"version\": \"1.2.0\",\n    \"environment\": \"development\"\n  }\n}"
    }
  ]
}
```

- **Quantity**: The minimum requirement is 10 examples, but more diverse and high-quality examples yield better results.
- **Quality**: Ensure your data accurately represents the task you want the model to perform.

!!! tip "Data preparation"
    For a detailed walkthrough of data preparation, see the [Fine-tuning Sentiment Analysis Tutorial](https://docs.kluster.ai/tutorials/klusterai-api/finetuning-sent-analysis/#get-the-data){target=_blank}.

!!! example "Find Llama datasets on Hugging Face"
    There is a wide range of datasets suitable for Llama model fine-tuning on [Hugging Face Datasets](https://huggingface.co/datasets?sort=trending&search=llama){target=_blank}. Browse trending and community-curated datasets to accelerate your data preparation.

> ### Set up the client

Reviewer: Ideally, we create a single file and then reference lines of the file. The idea is that we then at then end put the file to that people can copy paste

Reviewer 2: Done

---
title: Fine-tuning with the kluster.ai API
description: Learn how to programmatically create custom models tailored to your specific tasks by fine-tuning foundation models with your own data using the kluster.ai API.
---

# Fine-tuning with the kluster.ai API

The kluster.ai API lets you automate and integrate fine-tuning into your development workflows. You can create, manage, and monitor fine-tuning jobs directly from your code, making it easy to customize models for your specific needs.

This guide provides a practical overview of the fine-tuning process using the API. It covers the required data format, how to upload your dataset, and how to launch and monitor a fine-tuning job. For a step-by-step walkthrough, see the linked tutorial in the tips below.

## Prerequisites

Before getting started with fine-tuning, ensure you have the following:

> --8<-- 'text/kluster-api-onboarding.md'

Reviewer 2: Done!

- **Prepared dataset**: You need data formatted according to kluster.ai's requirements for fine-tuning (detailed below).

## Supported models

> kluster.ai currently supports fine-tuning for two base models:

Reviewer: ```suggestion
kluster.ai currently supports fine-tuning for the following models:
```

Reviewer: So it is more future-proof and we can just add models to the snippet

--8<-- 'text/get-started/fine-tuning-supported-models.md'

!!! info

>     You can query the [models endpoint](https://docs.kluster.ai/api-reference/reference/#list-supported-models){target=_blank} in the API and filter for the tag "fine-tunable."

Reviewer 3: ```suggestion
    You can query the [models endpoint](/api-reference/reference/#list-supported-models){target=_blank} in the API and filter for the tag "fine-tunable."
```
